libhdf5/nc4hdf.c

```diff
@@ -612,31 +611,59 @@ put_att_grpa(NC_GRP_INFO_T *grp, int varid, NC_ATT_INFO_T *att)
             BAIL(NC_EATTMETA);
 
         /* How big is the attribute? */
-        if ((existing_spaceid = H5Aget_space(existing_attid)) < 0)
+        if (att->nc_typeid == NC_CHAR) {
+          /* For text attributes the size is specified in the datatype
+             and it is enough to compare types using H5Tequal(). Here
+             we set npoints to disable the second part of the
+             comparison below. */
+          npoints = att->len;
+        }
+        else
+        {
+          if ((existing_spaceid = H5Aget_space(existing_attid)) < 0)
             BAIL(NC_EATTMETA);
-        if ((npoints = H5Sget_simple_extent_npoints(existing_spaceid)) < 0)
+          if ((npoints = H5Sget_simple_extent_npoints(existing_spaceid)) < 0)
             BAIL(NC_EATTMETA);
+        }
 
-        /* Delete the attribute. */
-        if (file_typeid != existing_att_typeid || npoints != att->len)
+        if (!H5Tequal(file_typeid, existing_att_typeid) || npoints != att->len)
```

Comment:
I should modify this condition instead of setting `npoints` to `att->len` above. This has been a long day...

Reply:
OK, it's a new day! ;-) Make the modification if you think it should be made.

Reply:
Done. I think this version makes it easier to see what's going on.